feature/google/src/test/java/com/revenuecat/purchases/google/BillingWrapperTest.kt

```diff
@@ -42,7 +61,87 @@ import java.util.concurrent.CountDownLatch
 
 @RunWith(AndroidJUnit4::class)
 @Config(manifest = Config.NONE)
-class BillingWrapperCommonTest : BillingWrapperTestBase() {
+class BillingWrapperTest {
+
+    private var onConnectedCalled: Boolean = false
+    private var mockClientFactory: BillingWrapper.ClientFactory = mockk()
+    private var mockClient: BillingClient = mockk()
+    private var purchasesUpdatedListener: PurchasesUpdatedListener? = null
+    private var billingClientStateListener: BillingClientStateListener? = null
+    private var handler: Handler = mockk()
+    private var mockDeviceCache: DeviceCache = mockk()
+
+    private var mockPurchasesListener: BillingAbstract.PurchasesUpdatedListener = mockk()
+
+    private var capturedAcknowledgeResponseListener = slot<AcknowledgePurchaseResponseListener>()
+    private var capturedAcknowledgePurchaseParams = slot<AcknowledgePurchaseParams>()
+    private var capturedConsumeResponseListener = slot<ConsumeResponseListener>()
+    private var capturedConsumeParams = slot<ConsumeParams>()
+
+    private lateinit var wrapper: BillingWrapper
+
+    private lateinit var mockDetailsList: List<SkuDetails>
+
+    private var storeProducts: List<StoreProduct>? = null
+
+    private val billingClientOKResult = BillingClient.BillingResponseCode.OK.buildResult()
+    private val appUserId = "jerry"
+    private var mockActivity = mockk<Activity>()
+
+    private val subsGoogleProductType = ProductType.SUBS.toGoogleProductType()!!
+    private val inAppGoogleProductType = ProductType.INAPP.toGoogleProductType()!!
+
+    @Before
+    fun setup() {
+        mockRunnables()
+
+        val listenerSlot = slot<PurchasesUpdatedListener>()
+        every {
+            mockClientFactory.buildClient(capture(listenerSlot))
+        } answers {
+            purchasesUpdatedListener = listenerSlot.captured
+            mockClient
+        }
+
+        val billingClientStateListenerSlot = slot<BillingClientStateListener>()
+        every {
+            mockClient.startConnection(capture(billingClientStateListenerSlot))
+        } answers {
+            billingClientStateListener = billingClientStateListenerSlot.captured
+        }
+
+        every {
+            mockClient.endConnection()
+        } just runs
+
+        every {
+            mockClient.acknowledgePurchase(
+                capture(capturedAcknowledgePurchaseParams),
+                capture(capturedAcknowledgeResponseListener)
+            )
+        } just Runs
```

Comment:
Nitpick, looks like we are using both `runs` and `Runs` in this file... We should go with just one of them. I would vote uppercase `Runs`, since the other is a typealias of the other but can go either way.

Reply:
honestly i always wondered what was going on there but kept forgetting to follow up. good call